Azure-Samples/chat-with-your-data-solution-accelerator
# IV resource creation

## Purpose

* Required by #321
* This PR adds helper classes for creating the Integrated Vectorization resources i.e index, indexer, skillset and datasource.
* It uses the existing `DocumentProcessor.process` method & checks if AZURE_SEARCH_USE_INTEGRATED_VECTORIZATION is enabled

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
If AZURE_SEARCH_USE_INTEGRATED_VECTORIZATION is True, when the user uploads a document, the search index, indexer, datasource & skillset should be created.
```

## What to Check
Verify that the following are valid
* Verify that the resources are created via the azure portal
* Verify that the document was indexed in the azure portal search resource
